src/H5FDros3.c

```diff
@@ -949,6 +1179,16 @@ H5FD__ros3_cmp(const H5FD_t *_f1, const H5FD_t *_f2)
     else if (f2->fa.secret_key[0] != '\0')
         HGOTO_DONE(-1);
 
+    /* FAPL: SESSION_TOKEN */
+    if (f1->fa.session_token[0] != '\0' && f2->fa.session_token[0] != '\0') {
```

Comment:
I don't think it necessarily makes sense to compare two files based on a session token and would generally be in favor of removing the `session_token` field from the structure and just passed around the value retrieved from the property.